### ISSUE TITLE
Add an option to let midhook call original first instead of calling hook code first

### ISSUE THREAD

**Issue by Nukoooo** [cursey/safetyhook#37]

**Comment by Patoke** [comment]
What you're trying to propose isn't intuitive in on itself, it would cause so much issues for no reason and the feature is entirely useless as you have already shown, you can just hook after the instruction was executed lol
If you need a certain instruction ran before your hook is executed, placing the hook after that instruction is executed is in fact intuitive since code flow goes as:
```asm
original_fn:
  mov rcx, [rax+330h]
  jmp hook_proxy
  ; continue_code_flow

hook_proxy:
  jmp hooked_fn
  ; the instructions after this is what is executed after you call original in your hook
  mov edx, 8003h ; original instructions that the library replaced
  jmp continue_code_flow
```
(also this isn't limited to mid function hooks XD)

**Comment by Nukoooo** [comment]
You are right about that. I saw the hooking library that [Dalamud](https://github.com/goatcorp/Dalamud) is using, found that the library has an option to [let you choose the behavior](https://github.com/Reloaded-Project/Reloaded.Hooks/blob/master/docs/AssemblyHooks.md#discarding-original-code) and thought that was useful in some cases, hence I made this issue.

**Comment by angelfor3v3r** [comment]
At the moment, we have to be careful with how MidHook is used. It currently doesn't back up registers outside flags & base x86 registers so it could easily crash. I believe cursey is working on fixing that (no promises on when it'll be done) and will probably take a look at this idea once that is done.

**Comment by angelfor3v3r** [comment]
Closing this for now.
